packages/web/src/features/translation/TranslateWord.tsx

```diff
@@ -15,91 +15,106 @@ export interface TranslateWordProps {
   originalLanguage: 'hebrew' | 'greek';
   status: 'empty' | 'saving' | 'saved';
   onGlossChange(gloss?: string): void;
+  onKeyDown?: KeyboardEventHandler<HTMLElement>;
 }
 
-export default function TranslateWord({
-  editable = false,
-  word,
-  originalLanguage,
-  status,
-  gloss,
-  referenceGloss,
-  previousGlosses,
-  onGlossChange,
-}: TranslateWordProps) {
-  const { t } = useTranslation();
-  const [text, setText] = useState(gloss ?? '');
-  const width = useTextWidth(text);
+const TranslateWord = forwardRef<HTMLInputElement, TranslateWordProps>(
```

Comment:
This component represents a group of controls, rather than a single input, so I think we should have its ref be our own interface. We can call it `TranslateWordRef` and have it just provide a focus function. We don't really want to expose the actual input element because that's an individual element of this component. This is where the hook `useImperativeHandle` lets us expose a custom object as the ref.

That's contrasted with the autocomplete input where the input element better represents the component and all of the other stuff augments it to make the input more powerful.

Reply:
Improved